src/babel-serializer/file.hh

```diff
@@ -13,15 +14,19 @@ namespace babel::file
 /// returns true if the file exists and can be read
 bool exists(cc::string_view filename);
 
-/// reads a file writes all bytes in the provided stream
+/// reads a file and writes all bytes in the provided stream (buffered)
 void read(cc::stream_ref<std::byte> out, cc::string_view filename, error_handler on_error = default_error_handler);
 
-/// reads a file and returns the content as string
+/// reads a file and returns the content as a string
 cc::string read_all_text(cc::string_view filename, error_handler on_error = default_error_handler);
 
-/// reads a file and returns the content as array of bytes
+/// reads a file and returns the content as an array of bytes
 cc::array<std::byte> read_all_bytes(cc::string_view filename, error_handler on_error = default_error_handler);
 
+/// reads a file and returns the content as an allocator-backed array of chars (null terminated)
+cc::alloc_array<char> read_all_text_alloc(char const* filename, cc::allocator* alloc, error_handler on_error = default_error_handler);
+cc::alloc_array<char> read_all_text_alloc(cc::string_view filename, cc::allocator* alloc, error_handler on_error = default_error_handler);
```

Comment:
we might even just name them `read_all_text` (overloads to the non-alloc version). Also, for consistency, how about an alloc version of read_all_bytes?